components/renderers/admin/AdminRenderer.js

```diff
@@ -25,6 +26,19 @@ const AdminRenderer = ({ type, ...params }) => {
       return <TextAreaRenderer type={type} {...params} />;
     case CMS_TYPES.FLOAT:
       return <FloatRenderer type={type} {...params} />;
+    case CMS_TYPES.CUSTOM:
+      const Component = plugin(params.customName);
+
+      // Can also cater to class components
+      if (Component.prototype?.render) {
+        // Class components would need this.props = props in
+        // their constructor
+        const instance = new Component({ type, ...params });
+
+        return instance.render();
+      }
+
+      return <Component type={type} {...params} />
```

Comment:
The return of `plugin` is not a React Component, but an ES6 class that encapsulates all of the behavior of the custom attribute type, from saving the database, to rendering in the API, and rendering in the Admin part. Sorry if I confused you with using the `render()` method name - this is not intended to be a React Class component, but a normal class that, when instantiated, returns a React component.

It would something along the lines of:

```
      const Plugin = plugin(params.customName);
      const plugin = new Plugin();
      plugin.toJSON(); //handles the rendering of the API;
      plungin.renderAdmin(); //returns a React component that handles the admin side
      plugin.save(); //handles the logic needed to save the database
```


```suggestion
      const Component = plugin(params.customName);
      const component = new Component();
      return component.renderAdmin(type, params);   //returns a React component
```